### DISCUSSION TITLE
Converting typst to LaTeX: can I tell pandoc to "pass through" certain commands?

### DISCUSSION THREAD

**Discussion by nullst** [jgm/pandoc#9929] [Q&A]
I want to use `pandoc` to convert a typst document I wrote to LaTeX. Ideally I want the resulting LaTeX code to be "human-friendly", as in, I intend to keep working on the produced LaTeX document (mostly editing, but I may need to add substantial new content), discarding the typst source after the conversion as no longer relevant.

Is there a way to tell `pandoc` to treat certain typst commands as black boxes, just converting the syntax instead of substituting the definition? An example: I want the typst fragment
```typst
#theorem[
  Main theorem.
]
```
to be translated into
```latex
\begin{theorem}
  Main theorem.
\end{theorem}
```
Here the typst source uses the `ctheorems` package and defines the `#theorem` command using the functions exported from that package, but I explicitly don't want pandoc to look deep into what "theorem" means and how it is defined (I'm not 100% sure that `pandoc` can actually do that), I want the tautological translation. Similarly, my typst code uses many math commands like `Dbcoh` that I would like to be translated just into `\Dbcoh` regardless of what the definition is.

Judging by what I understand the `pandoc` philosophy to be, this is probably not possible, but maybe there is a way?

P.S. To be honest, I have already solved my current problem by writing a python script that performs a bunch of regular expression substitutions to convert my typst code to, like, 98% correct LaTeX code and manually fixed all remaining issues, but this only worked because I treat typst purely as a level of syntax sugar on top of LaTeX and adhere to the same writing style. I would like a more robust solution.

**Comment by jgm** [comment]
I can't think of any way to achieve that with pandoc -- sorry.

**Comment by aualbert** [comment]
@nullst I have started building a human-friendly Typst to Latex converter that does just that! See https://github.com/aualbert/typst2latex. It uses pandoc as a backend.
